apps/marketplace/pages/SellerNotifyPage.js

```diff
@@ -0,0 +1,263 @@
+import SellerNotifySelect from 'marketplace/components/SellerNotify/SellerNotifySelect'
+import SellerNotifyReview from 'marketplace/components/SellerNotify/SellerNotifyReview'
+import SellerNotifySellerList from 'marketplace/components/SellerNotify/SellerNotifySellerList'
+import ErrorBox from 'shared/form/ErrorBox'
+import { rootPath } from 'marketplace/routes'
+import { loadBriefSellers, handleBriefSellerNotifySubmit } from 'marketplace/actions/briefActions'
+import LoadingIndicatorFullPage from 'shared/LoadingIndicatorFullPage/LoadingIndicatorFullPage'
+
+class SellerNotifyPage extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      // this is the current state of the nav items in the progress indictor
+      stages: {
+        introduction: 'todo',
+        select: 'todo',
+        review: 'todo'
+      },
+      // this is the state of whether the nav items are done
+      stagesDone: {
+        introduction: false,
+        select: false,
+        review: false
+      },
+      selectedSellers: []
+    }
+    this.setStageStatus = this.setStageStatus.bind(this)
+    this.moveToNextStage = this.moveToNextStage.bind(this)
+    this.moveToStage = this.moveToStage.bind(this)
+    this.selectSeller = this.selectSeller.bind(this)
+    this.deselectSeller = this.deselectSeller.bind(this)
+    this.hasSelectedASeller = this.hasSelectedASeller.bind(this)
+    this.setStageDoneStatus = this.setStageDoneStatus.bind(this)
+  }
+
+  componentDidMount() {
+    this.props.loadInitialData(this.props.match.params.briefId)
+  }
+
+  setStageStatus(stage, status) {
+    this.setState(currentState => {
+      const newState = { ...currentState }
+      if (status === 'doing') {
+        // there can only be one "doing" stage
+        Object.keys(newState.stages).map(stateStage => {
+          if (newState.stages[stateStage] === 'doing') {
+            newState.stages[stateStage] = currentState.stagesDone[stateStage] ? 'done' : 'todo'
+          }
+          return true
+        })
+      }
+      newState.stages[stage] = status
+      return newState
+    })
+  }
+
+  setStageDoneStatus(stage, isDone) {
+    this.setState(currentState => {
+      const newState = { ...currentState }
+      newState.stagesDone[stage] = isDone
+      return newState
+    })
+  }
+
+  moveToNextStage(currentStage) {
+    const stages = Object.keys(this.state.stages)
+    const index = stages.indexOf(currentStage)
+
+    if (index !== -1 && typeof stages[index + 1] !== 'undefined') {
+      this.props.history.push(
+        `${rootPath}/brief/${this.props.match.params.briefId}/seller-${this.props.match.params.flow}/${stages[
+          index + 1
+        ]}`
+      )
+    }
+  }
+
+  moveToStage(stage) {
+    if (this.state.stages[stage]) {
+      this.props.history.push(
+        `${rootPath}/brief/${this.props.match.params.briefId}/seller-${this.props.match.params.flow}/${stage}`
+      )
+    }
+  }
+
+  selectSeller(supplierCode, callback) {
```

Comment:
It's my understanding redux should be used where propagating stage changes would involve intermediate components that do nothing but pass the state down to deeper child components. In this case selecting a seller (and other similar functions) is only needed by direct children of the page component so it may be overkill to involve redux. Currently the child components do not need to be connected to the redux store.